engine/engine.go

```diff
@@ -145,6 +146,7 @@ func New(opts Opts) *compatibilityEngine {
 
 type compatibilityEngine struct {
 	prom    *promql.Engine
+	timeout time.Duration
```

Comment:
Let's maybe move the timeout bellow the `logicalOptimizers` variable to group query parameters in one place?